src/plots/cartesian/set_convert.js

```diff
@@ -405,10 +405,15 @@ module.exports = function setConvert(ax, fullLayout) {
         if(axLetter in trace) {
             arrayIn = trace[axLetter];
             len = trace._length || arrayIn.length;
-            arrayOut = new Array(len);
 
-            for(i = 0; i < len; i++) {
-                arrayOut[i] = ax.d2c(arrayIn[i], 0, cal);
+            if(ax.type === 'linear' && Lib.isTypedArray(arrayIn) && arrayIn.subarray) {
+                arrayOut = arrayIn.subarray(0, len);
+            } else {
+                arrayOut = new Array(len);
```

Comment:
In cases like this where we know the length and we know we're dealing with numbers, would we benefit by using typed arrays internally? When they're supported of course - we'd need a helper to revert to `Array` in IE9. And subject to my [question about `NaN`](https://github.com/plotly/plotly.js/pull/2388/commits/45c2f35ac2332a78147b7238b9a2847115192254#r171131109).

For a later time regardless...

Reply:
That's an idea. Saving a few `isNumeric` (assuming `isNaN` is faster) calls downstream of the `makeCalcdata` should help a little bit.